server/builds.go

```diff
@@ -39,23 +39,22 @@ func (b *Builds) dockerRegistryClient(s *Server) (reg *registry.Registry, err er
 	return reg, nil
 }
 
-func (b *Builds) waitForImage(ctx context.Context, s *Server, reg *registry.Registry, pr *model.PullRequest, imageToCheck string, logger logrus.FieldLogger) (*model.PullRequest, error) {
-	desiredTag := b.getInstallationVersion(pr)
+func (b *Builds) waitForImage(ctx context.Context, reg *registry.Registry, desiredTag, imageToCheck string, logger logrus.FieldLogger) error {
 	logger = logger.WithFields(logrus.Fields{"image": imageToCheck, "tag": desiredTag})
 
 	for {
 		select {
 		case <-ctx.Done():
-			return pr, errors.New("timed out waiting for image to publish")
+			return errors.New("timed out waiting for image to publish")
```

Comment:
Should `waitForImage` handle its own timeout? Like... let's allow 20m for a image to be created but a global timeout of 45m for the entire process on the parent context. (?)

Just exposing what comes to mind. No need to make any changes.

Reply:
Yeah, that probably makes sense to do here. I will create a ticket so we remember to add it in our next improvement. Thanks!